### PR TITLE
Add support for ARM64

### PR DESCRIPTION
**Pull request by mikeeq**

### CODE REVIEW

File: playbooks/fedora.yml [hunk 1]
```diff
@@ -8,15 +8,15 @@
     # - debug:
     #     var: hostvars[inventory_hostname]
   roles:
-    - pkg_mgmt/dnf
```
Comment (mikeeq) [inline playbooks/fedora.yml:11]: please uncomment and test again

File: playbooks/generic.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@
   hosts: "{{ hosts_to_deploy | default('local') }}"
   become: true
   roles:
-    - pkg_mgmt/pip
```
Comment (mikeeq) [inline playbooks/generic.yml:6]: please uncomment and test again, paste output from azure cli to see that it's working

```
az version
```

File: playbooks/group_vars/all.yml [hunk 3]
```diff
@@ -101,14 +101,14 @@ flatpak_apps:
   #   remote: flathub
   # - name: org.signal.Signal
   #   remote: flathub
-  - name: com.spotify.Client
```
Comment (mikeeq) [inline playbooks/group_vars/all.yml:104]: put it as a separate group_vars for x86_64 and arm64

File: roles/apps/kubectl/tasks/krew.yml [hunk 4]
```diff
@@ -26,3 +26,4 @@
   when: installed_krew_version.stdout != krew_version
 
 - import_tasks: krew/plugins.yml
+  when: ansible_architecture != "aarch64"
```
Comment (mikeeq) [inline roles/apps/kubectl/tasks/krew.yml:29]: remove, add `include_vars` like in packer role i.e.

File: roles/apps/kubectl/tasks/main.yml [hunk 5]
```diff
@@ -1,10 +1,17 @@
 ---
+
 - name: Get installed version
   shell: kubectl version --short --client=true | grep -i client | cut -d'v' -f2
   register: installed_kubectl_version
   changed_when: false
   ignore_errors: true
 
+- include_vars: x86_64.yml
+  when: ansible_architecture == "x86_64" or "amd64"
```
Comment (mikeeq) [inline roles/apps/kubectl/tasks/main.yml:10]: remove or

File: roles/apps/kubectl/vars/x86_64.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+package_name: "amd64"
+
+# https://github.com/ishantanu/awesome-kubectl-plugins
+krew_plugins:
+  - sniff
+  - hns
```
Comment (mikeeq) [inline roles/apps/kubectl/vars/x86_64.yml:7]: test if it's working on arm64

File: roles/apps/packer/tasks/main.yml [hunk 7]
```diff
@@ -12,5 +12,11 @@
 - include_tasks: get_latest_version.yml
   when: packer_version is not defined
 
+- include_vars: x86_64.yml
+  when: ansible_architecture == "x86_64" or ansible_architecture == "amd64"
```
Comment (mikeeq) [inline roles/apps/packer/tasks/main.yml:16]: remove or

File: roles/apps/vagrant/tasks/main.yml [hunk 8]
```diff
@@ -1,70 +1,4 @@
...
-# # - name: Install vagrant from fedora repo
-# #   dnf:
-# #     name: "{{ pkgs }}"
-# #     state: latest
-# #   vars:
-# #     pkgs:
-# #       - vagrant
-# #       - vagrant-libvirt
+- include_tasks: install.yml
+  when: ansible_architecture != "aarch64"
```
Comment (mikeeq) [inline roles/apps/vagrant/tasks/main.yml:4]: add a comment above that vagrant is not available for arm64 linux

File: playbooks/group_vars/all.yml [hunk 9]
```diff
@@ -1,5 +1,5 @@
 ---
-user_name: 'mikee'
```
Comment (mikeeq) [inline playbooks/group_vars/all.yml:2]: revert

File: roles/desktop/gnome/tasks/main.yml [hunk 10]
```diff
@@ -4,3 +4,4 @@
 - import_tasks: gsettings.yml
 - import_tasks: extensions_manual.yml
 - import_tasks: remote_desktop.yml
+  # when:
```
Comment (mikeeq) [inline roles/desktop/gnome/tasks/main.yml:7]: remove/rever

File: playbooks/fedora.yml [hunk 11]
```diff
@@ -8,20 +8,26 @@
     # - debug:
     #     var: hostvars[inventory_hostname]
   roles:
+    - role: pkg_mgmt/flatpak
+      when: ansible_architecture == "x86_64"
+      import_vars: ./group_vars/x86_64.yml
```
Comment (Kobeep) [inline playbooks/fedora.yml:13]: It doesn't rely on x86_64.yml group, still searching in all.yml.

**Pull request merged**